src/main/kotlin/org/opensearch/commons/alerting/model/DataSources.kt

```diff
@@ -89,6 +86,28 @@ data class DataSources(
         findingsEnabled = sin.readOptionalBoolean()
     )
 
+    constructor(
```

Comment:
https://github.com/opensearch-project/common-utils/pull/674/files#diff-97cf101698aba14b5844f08b1d5be6fcf2752671b9d7510d1a566efbeec20793R83

comments index should be using readOptionalString(), writeOptionalString, null check in parsing etc. 
plz incorporate

Reply:
updating readOptional and writeOptional, parse adds an empty string placeholder for any comments that don't exist in the XContent, so leaving that be